pkg/utils/cachedClient/get_cached_client.go

```diff
@@ -17,12 +18,14 @@ import (
 
 // GetCachedClient returns a controller runtime client with the cache initialized only for the resources added to
 // the scheme.
-func GetCachedClient(ctx context.Context, scheme *runtime.Scheme) (client.Client, error) {
-	conf := ctrl.GetConfigOrDie()
+func GetCachedClient(ctx context.Context, scheme *runtime.Scheme, conf *rest.Config) (client.Client, error) {
```

Comment:
Nit. Probably it is more idiomatic creating a second method `GetCachedClientWithConfig` (which this one relies upon) instead of relying on the implicit fact conf is null to create a new config, as it may lead to unexpected bugs (e.g., if you pass a variable which is null by mistake).

Reply:
great idea, thank you.